tests/back/api/test_experiments_api.py

```diff
@@ -2,111 +2,138 @@
+
+    # test get experiment by id 1.
     response = client.get("/api/v1/experiment/1")
-    assert response.status_code == 200, response.text
+    assert response.status_code == 200
     data = response.json()
     assert data["dataset_id"] == dataset_id
     assert data["task_name"] == "TabularClassificationTask"
     assert data["name"] == "ExperimentA"
 
+    # test get experiment by id 2.
     response = client.get("/api/v1/experiment/2")
-    assert response.status_code == 200, response.text
+    assert response.status_code == 200
     data = response.json()
     assert data["dataset_id"] == dataset_id
     assert data["task_name"] == "TabularClassificationTask"
     assert data["name"] == "Experiment2"
 
 
 def test_get_all_experiments(client: TestClient, dataset_id: int):
-    # Get all the experiments available in the back
+    """Test that all experiments can be retrieved."""
     response = client.get("/api/v1/experiment")
-    assert response.status_code == 200, response.text
+
+    assert response.status_code == 200
+
     data = response.json()
+    assert len(data) == 2
     assert data[0]["dataset_id"] == dataset_id
     assert data[1]["dataset_id"] == dataset_id
 
 
-def test_get_wrong_experiment(client: TestClient):
-    # Try to retrieve a non-existent experiment an get an error
+def test_not_found_experiment(client: TestClient):
+    """Test that a 404 is returned when the experiment is not found."""
     response = client.get("/api/v1/experiment/31415")
-    assert response.status_code == 404, response.text
+
+    assert response.status_code == 404
     assert response.text == '{"detail":"Experiment not found"}'
 
 
-def test_modify_experiment(client: TestClient, dataset_id: int):
-    # Modify an existent experiment
+def test_update_experiment(client: TestClient, dataset_id: int):
+    """Test that an experiment can be updated through a patch call."""
+
     response = client.patch(
         "/api/v1/experiment/2?task_name=UnknownTask&name=Experiment123",
     )
-    assert response.status_code == 200, response.text
+    assert response.status_code == 200
 
-    # Get the experiment
+    # get the updated experiment
     response = client.get("/api/v1/experiment/2")
-    assert response.status_code == 200, response.text
+    assert response.status_code == 200
+
     data = response.json()
     assert data["dataset_id"] == dataset_id
     assert data["task_name"] == "UnknownTask"
     assert data["name"] == "Experiment123"
     assert data["created"] != data["last_modified"]
 
 
-def test_modify_experiment_step(client: TestClient):
+def test_update_experiment_step(client: TestClient):
+    """Test that an experiment step can be updated through a patch call."""
     response = client.patch(
         "/api/v1/experiment/2",
         data={"params": """{"step": "STARTED"}""", "url": ""},
     )
-    assert response.status_code == 304, response.text
+    assert response.status_code == 304
```

Comment:
If you are looking for more detailed information for debugging when a test fails, using assert response.status_code == 200, response.text is preferable. It gives immediate insights into why the response did not meet the expected status code, which can save time during the development and debugging process